bottlecap/src/proc/mod.rs

```diff
@@ -0,0 +1,82 @@
+pub mod constants;
+
+use std::{
+    error::Error,
+    fs::File,
+    io::{self, BufRead},
+};
+
+use constants::{LAMDBA_NETWORK_INTERFACE, PROC_NET_DEV_PATH};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct NetworkData {
+    pub rx_bytes: f64,
+    pub tx_bytes: f64,
+}
+
+pub fn get_network_data() -> Result<NetworkData, Box<dyn std::error::Error>> {
+    get_network_data_helper(PROC_NET_DEV_PATH)
+}
+
+fn get_network_data_helper(path: &str) -> Result<NetworkData, Box<dyn std::error::Error>> {
+    let file = File::open(path).map_err(|e| Box::new(e) as Box<dyn Error>)?;
```

Comment:
Why do we need to box the error?

Reply:
You're right, it isn't needed! Changing it to io::Error